syntax = "proto3";
package rosbridge_service;

import public "google/protobuf/empty.proto";

option java_multiple_files = true;
option java_package = "org.athenian.grpc";

message TwistData {
    double linear_x = 1;
    double linear_y = 2;
    double linear_z = 3;
    double angular_x = 4;
    double angular_y = 5;
    double angular_z = 6;
}

message EncoderDesc {
    string name = 1;
}

message EncoderData {
    double value = 1;
}

service RosBridgeService {
    rpc writeTwistData (TwistData) returns (google.protobuf.Empty) {
    }

    rpc streamTwistData (stream TwistData) returns (google.protobuf.Empty) {
    }

    rpc readEncoderData (EncoderDesc) returns (stream EncoderData) {
    }
}
